fieldtracing/fieldtracing.cpp

```diff
@@ -996,114 +1010,213 @@ namespace FieldTracing {
+                     cellsToDoFluxRopes++;
+                  }
                }
-               if(reducedCellNeedsContinuedBWTracing[n] > 0) {
-                  anyCellNeedsTracing=true;
-                  cellNeedsContinuedBWTracing[n] = 1;
-                  cellsToDo++;
-                  
+               if(cellNeedsContinuedBWTracing[n] > 0) {
+                  cellsToDoFullBox++;
                   // Update that nodes' tracing coordinates
                   cellBWTracingCoordinates[n][0] = sumCellBWTracingCoordinates[n][0] / reducedCellNeedsContinuedBWTracing[n];
                   cellBWTracingCoordinates[n][1] = sumCellBWTracingCoordinates[n][1] / reducedCellNeedsContinuedBWTracing[n];
                   cellBWTracingCoordinates[n][2] = sumCellBWTracingCoordinates[n][2] / reducedCellNeedsContinuedBWTracing[n];
-                  
-                  cellBWStepCounter[n] = maxCellBWStepCounter[n];
+                  if(cellBWConnection[n] == TracingLineEndType::UNPROCESSED) {
+                     cellsToDoFluxRopes++;
+                  }
                }
-               cellFWTracingStepSize[n] = reducedCellFWTracingStepSize[n];
-               cellBWTracingStepSize[n] = reducedCellBWTracingStepSize[n];
             }
             #pragma omp barrier
-         } while(anyCellNeedsTracing && (cellsToDo >= fieldTracingParameters.max_incomplete_lines_fullbox * 2 * globalDccrgSize));
+         } while(!(
+            cellsToDoFullBox <= fieldTracingParameters.fullbox_max_incomplete_lines * 2 * globalDccrgSize
+            && cellsToDoFluxRopes <= fieldTracingParameters.fluxrope_max_incomplete_lines * 2 * globalDccrgSize
+         ));
          
          // Last pass to sort cells that would still continue but won't as we exited as TracingLineEndType::LOOP instead of UNPROCESSED.
          #pragma omp for schedule(dynamic)
          for(int n=0; n<globalDccrgSize; n++) {
             if(cellNeedsContinuedFWTracing[n] == 1) {
-               cellFWConnection[n] = TracingLineEndType::LOOP;
+               reducedCellFWConnection[n] += TracingLineEndType::LOOP;
             }
             if(cellNeedsContinuedBWTracing[n] == 1) {
-               cellBWConnection[n] = TracingLineEndType::LOOP;
+               reducedCellBWConnection[n] += TracingLineEndType::LOOP;
             }
          }
       } // pragma omp parallel
       phiprof::stop("loop");
       
-      logFile << "(fieldtracing) full box tracing traced in " << itCount << " iterations of the tracing loop with " << cellsToDo << " remaining incomplete field lines (total spatial cells " << globalDccrgSize <<  ")." << endl;
+      logFile << "(fieldtracing) combined flux rope + full box tracing traced in " << itCount
+         << " iterations of the tracing loop with flux rope " << cellsToDoFluxRopes
+         << ", full box " << cellsToDoFullBox
+         << " remaining incomplete field lines (total spatial cells " << globalDccrgSize
+         << ")." << endl;
```

Comment:
As soon as we are fully confident that the tracing works to our satisfaction, we could put this one behind a "DEBUG" define as well.
But obviously, that is not yet so. :)

Reply:
I think it's useful to have some idea of how it behaves at runtime. If we see insane amounts of iterations we'll want to change parameters.